nodejs/week2/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,107 @@
+const { query, request } = require("express");
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+//...............Respond with the json for the meal with the corresponding id
+
+router.get("/:id", async (request, response) => {
```

Comment:
remember you can use a regexp in the router, i.e. `router.get("/:id(\\d+)", (req, res) => {...}` This will make your life a lot easier if you want to enforce integer id's :)

Reply:
If you do something like that you will not need all the below validation code.